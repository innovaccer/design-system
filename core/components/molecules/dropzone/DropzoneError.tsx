import * as React from 'react';
import { Text, Icon } from '@/index';
import { DropzoneProps } from '@/index.type';
import classNames from 'classnames';

export interface DropzoneErrorProps {
  error: string;
  type: DropzoneProps['type'];
}

export const DropzoneError = (props: DropzoneErrorProps) => {
  const { type, error } = props;

  const IconClass = classNames({
    ['Dropzone-stateIcon']: true,
    [`Dropzone-stateIcon--${type}`]: type,
  });

  return (
    <React.Fragment>
      {type !== 'tight' && <Icon name="error" size={64} appearance="alert" className={IconClass} />}
      <Text appearance="destructive" size="large" weight="strong">
        {error}
      </Text>
    </React.Fragment>
  );
};

DropzoneError.displayName = 'DropzoneError';

export default DropzoneError;
