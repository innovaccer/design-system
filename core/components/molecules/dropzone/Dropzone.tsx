import * as React from 'react';
import classNames from 'classnames';
import { Text } from '@/index';
import DropzoneBase, { DropzoneBaseProps } from './DropzoneBase';
import DropzoneActive from './DropzoneActive';
import DropzoneError from './DropzoneError';
import { BaseProps, extractBaseProps } from '@/utils/types';
import { fileErrorMessages } from './FileErrors';
import { useAccessibilityProps } from '@/accessibility/utils';
import DropzoneIcon from './DropzoneIcon';
import styles from '@css/components/dropzone.module.css';

export type DropZoneType = 'standard' | 'compressed' | 'tight';

export interface DropzoneProps extends BaseProps, DropzoneBaseProps {
  /**
   * Description of accepted formats in `FileUploader`
   */
  formatLabel?: string;
  /**
   * Size of `Dropzone`
   */
  type: DropZoneType;
  /**
   * Description of maximum size in `FileUploader`
   */
  sizeLabel?: string;
  /**
   * Link component to download sample file
   */
  sampleFileLink?: React.ReactNode;
}

export const Dropzone = (props: DropzoneProps) => {
  const { type, sizeLabel, className, formatLabel, sampleFileLink, disabled } = props;

  const { open, getRootProps, getInputProps, isDragActive, isDragReject, fileError } = DropzoneBase(props);

  const baseProps = extractBaseProps(props);

  const DropzoneClass = classNames(
    {
      [styles['Dropzone']]: true,
      [styles[`Dropzone--${type}`]]: type,
      [styles['Dropzone--disabled']]: disabled,
      [styles['Dropzone--active']]: isDragActive,
      [styles['Dropzone--error']]: isDragReject,
      [styles['Dropzone-animation']]: true,
      [styles['Dropzone-animation--default']]: !isDragActive && type !== 'standard',
      [styles['Dropzone-animation--active']]: isDragActive && !isDragReject && type !== 'standard',
      [styles['Dropzone-standard--default']]: !isDragActive && type === 'standard',
      [styles['Dropzone-standard--active']]: isDragActive && !isDragReject && type === 'standard',
    },
    className
  );

  const WrapperClass = classNames({
    [styles['DropzoneWrapper']]: true,
    [styles[`DropzoneWrapper--${type}`]]: true,
  });

  const renderDropzone = () => {
    if (isDragReject) return <DropzoneError type={type} error={fileErrorMessages[fileError]} />;

    if (isDragActive) return <DropzoneActive type={type} />;

    const buttonAccessibilityProps = useAccessibilityProps({
      onClick: open,
      'aria-label': 'Drag your files here or click to select files',
    });

    return (
      <React.Fragment>
        {type !== 'tight' && <DropzoneIcon disabled={disabled} name="default" type={type} />}
        <div className={WrapperClass} data-test="DesignSystem-Dropzone-Wrapper">
          <span>
            <Text size="large" weight="strong" className="mr-2" appearance={disabled ? 'disabled' : 'default'}>
              Drag your files here or
            </Text>
            <Text
              className="ml-2 cursor-pointer"
              size="large"
              weight="strong"
              appearance={disabled ? 'disabled' : 'link'}
              {...buttonAccessibilityProps}
              tabIndex={disabled ? -1 : 0}
            >
              browse files
            </Text>
            <input {...getInputProps()} />
          </span>
          {formatLabel && <Text appearance={disabled ? 'disabled' : 'subtle'}>{formatLabel}</Text>}
          {sizeLabel && <Text appearance={disabled ? 'disabled' : 'subtle'}>{sizeLabel}</Text>}
          {sampleFileLink && <div className="mt-5">{sampleFileLink}</div>}
        </div>
      </React.Fragment>
    );
  };

  return (
    <div {...getRootProps()} {...baseProps} className={DropzoneClass} data-test="DesignSystem-Dropzone">
      {renderDropzone()}
    </div>
  );
};

Dropzone.displayName = 'Dropzone';

Dropzone.defaultProps = {
  ...DropzoneBase.defaultProps,
  type: 'standard',
};

export default Dropzone;
