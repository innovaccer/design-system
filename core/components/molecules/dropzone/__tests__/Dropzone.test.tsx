import { render, fireEvent } from '@testing-library/react';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import Dropzone, { DropzoneProps as Props, DropZoneType } from '../Dropzone';
import Link from '@/components/atoms/link';
import DropzoneError from '../DropzoneError';
import { fromEvent } from '../FileSelectorUtils';

const formatLabel = 'Accepted formats: PDF, JPG';
const sizeLabel = 'Maximum size: 30 MB';
const BooleanValue = [true, false];
const FunctionValue = jest.fn();

const fileData = {
  dataTransfer: {
    files: [new File(['(⌐□_□)'], 'file1.png')],
    types: ['Files'],
  },
  size: 1024,
};

const multipleFileData = {
  dataTransfer: {
    files: [new File(['(⌐□_□)'], 'file1.png'), new File(['(⌐□_□)'], 'file2.jpeg')],
    types: ['Files'],
  },
  size: 2048,
};

const minSize = 50;
const maxSize = 300;
const types: DropZoneType[] = ['standard', 'compressed', 'tight'];

describe('Dropzone component snapshot', () => {
  const mapper: Record<string, any> = {
    formatLabel: valueHelper(formatLabel, { required: true }),
    sizeLabel: valueHelper(sizeLabel, { required: true }),
    disabled: valueHelper(BooleanValue, { required: true, iterate: true }),
    type: valueHelper(types, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { asFragment } = render(<Dropzone {...attr} />);
      expect(asFragment()).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Dropzone component snapshot', () => {
  const mapper: Record<string, any> = {
    maxSize: valueHelper(maxSize, { required: true }),
    minSize: valueHelper(minSize, { required: true }),
    accept: 'image/png',
    preventDropOnDocument: valueHelper(BooleanValue, { required: true, iterate: true }),
    multiple: valueHelper(BooleanValue, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { asFragment } = render(<Dropzone {...attr} />);
      expect(asFragment()).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Dropzone component', () => {
  it('renders prop:formatLabel', () => {
    const { getAllByTestId } = render(<Dropzone formatLabel={formatLabel} />);
    expect(getAllByTestId('DesignSystem-Text')[2]).toHaveTextContent(formatLabel);
  });

  it('renders prop:sizeLabel', () => {
    const { getAllByTestId } = render(<Dropzone sizeLabel={sizeLabel} />);
    expect(getAllByTestId('DesignSystem-Text')[2]).toHaveTextContent(sizeLabel);
  });

  it('renders prop:sizeLabel & prop:formatLabel', () => {
    const { getAllByTestId } = render(<Dropzone sizeLabel={sizeLabel} formatLabel={formatLabel} />);
    expect(getAllByTestId('DesignSystem-Text')[2]).toHaveTextContent(formatLabel);
    expect(getAllByTestId('DesignSystem-Text')[3]).toHaveTextContent(sizeLabel);
  });

  it('check for prop:disabled', () => {
    const { getByTestId } = render(<Dropzone disabled={true} />);
    expect(getByTestId('DesignSystem-Dropzone')).toHaveClass('Dropzone--disabled');
  });
});

describe('Dropzone component with prop:type', () => {
  types.forEach((type) => {
    it(`should have the Dropzone--${type} class when type=${type} `, () => {
      const { getByTestId } = render(<Dropzone type={type} />);
      expect(getByTestId('DesignSystem-Dropzone')).toHaveClass(`Dropzone--${type}`);
      expect(getByTestId('DesignSystem-Dropzone-Wrapper')).toHaveClass(`DropzoneWrapper--${type}`);
    });
  });
});

describe('Dropzone component prop:sampleFileLink', () => {
  it('check for sampleFileLink element', () => {
    const { getByTestId } = render(
      <Dropzone
        sampleFileLink={
          <Link href="" download="Test.pdf" target="_blank">
            Download sample file
          </Link>
        }
      />
    );
    expect(getByTestId('DesignSystem-Link')).toBeInTheDocument();
  });
});

describe('Dropzone component event handlers', () => {
  it('checks for onDragLeave event', () => {
    const { getByTestId } = render(<Dropzone onDragLeave={FunctionValue} accept="image/png" />);
    fireEvent.dragLeave(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('checks for onDragEnter event', () => {
    const { getByTestId } = render(
      <Dropzone onDragLeave={FunctionValue} accept="image/png" maxSize={maxSize} minSize={minSize} multiple={false} />
    );
    fireEvent.dragEnter(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('checks for onDragOver event', () => {
    const { getByTestId } = render(<Dropzone onDragOver={FunctionValue} accept="image/png" />);
    fireEvent.dragOver(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('checks for onDrop event', () => {
    const { getByTestId } = render(<Dropzone onDrop={FunctionValue} accept="image/png" />);
    fireEvent.drop(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('checks for onDropAccepted event', () => {
    const { getByTestId } = render(<Dropzone onDropAccepted={FunctionValue} accept="image/png" />);
    fireEvent.drop(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('checks for onDropRejected	event', () => {
    const { getByTestId } = render(<Dropzone onDropRejected={FunctionValue} accept="image/jpeg" />);
    fireEvent.drop(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('check for browse file onOpenDialog option', () => {
    const { getAllByTestId } = render(<Dropzone onDropAccepted={FunctionValue} />);

    fireEvent.click(getAllByTestId('DesignSystem-Text')[1]);
    fireEvent.focus(window);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('check for browse file onCloseDialog option', () => {
    const { getAllByTestId, getByTestId } = render(
      <Dropzone onFileDialogCancel={FunctionValue} onDragEnter={FunctionValue} />
    );

    fireEvent.dragEnter(getByTestId('DesignSystem-Dropzone'), fileData);
    fireEvent.click(getAllByTestId('DesignSystem-Text')[1]);
    fireEvent.focus(window);
    fireEvent.keyDown(window, { key: 'Escape', code: 'Escape' });

    expect(FunctionValue).toHaveBeenCalled();
  });
});

describe('dropzone component prop:multiple', () => {
  it('renders component when multiple=false', () => {
    const { getByTestId } = render(<Dropzone onDropRejected={FunctionValue} multiple={false} />);

    fireEvent.drop(getByTestId('DesignSystem-Dropzone'), multipleFileData);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('renders component when multiple=true', () => {
    const { getByTestId } = render(<Dropzone onDrop={FunctionValue} multiple={true} />);

    fireEvent.drop(getByTestId('DesignSystem-Dropzone'), multipleFileData);

    expect(FunctionValue).toHaveBeenCalled();
  });
});

describe('dropzone component prop:accept', () => {
  it('checks when prop:accept is passed as an array', () => {
    const { getByTestId } = render(<Dropzone onDrop={FunctionValue} multiple={true} accept={['image/png']} />);

    fireEvent.drop(getByTestId('DesignSystem-Dropzone'), multipleFileData);

    expect(FunctionValue).toHaveBeenCalled();
  });
});

describe('dropzone component prop:Size', () => {
  it('renders component when minSize exceeds file size', () => {
    const { getByTestId } = render(
      <Dropzone
        onDropRejected={FunctionValue}
        maxSize={maxSize}
        minSize={minSize}
        multiple={false}
        accept="image/jpeg"
      />
    );

    fireEvent.drop(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('renders component when file size exceeds maxSize', () => {
    const { getByTestId } = render(
      <Dropzone
        onDropRejected={FunctionValue}
        maxSize={10}
        minSize={minSize + 2000}
        multiple={false}
        accept="image/jpeg"
      />
    );

    fireEvent.drop(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(FunctionValue).toHaveBeenCalled();
  });
});

describe('DropzoneError Component', () => {
  it('check for error message', () => {
    const { getByTestId } = render(<DropzoneError error="File Type Not Supported" type="standard" />);
    expect(getByTestId('DesignSystem-Text')).toHaveTextContent('File Type Not Supported');
  });
});

describe('dropzone component prop:getFilesFromEvent', () => {
  it('checks for onDragEnter event', () => {
    const { getByTestId } = render(
      <Dropzone
        onDragEnter={FunctionValue}
        multiple={true}
        accept={['image/png']}
        getFilesFromEvent={fromEvent}
        minSize={minSize}
        maxSize={maxSize}
      />
    );

    fireEvent.dragEnter(getByTestId('DesignSystem-Dropzone'), {});
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('checks when dragEnter rejected event', () => {
    const { getByTestId } = render(
      <Dropzone
        onDropRejected={FunctionValue}
        multiple={true}
        accept={['image/jpeg']}
        minSize={minSize}
        maxSize={maxSize}
      />
    );

    fireEvent.dragEnter(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('checks for onDrop event', () => {
    const { getByTestId } = render(
      <Dropzone
        onDrop={FunctionValue}
        multiple={true}
        accept={['image/png']}
        getFilesFromEvent={fromEvent}
        minSize={minSize}
        maxSize={maxSize}
      />
    );
    fireEvent.drop(getByTestId('DesignSystem-Dropzone'), fileData);
    expect(getByTestId('DesignSystem-Dropzone')).toBeInTheDocument();
    expect(FunctionValue).toHaveBeenCalled();
  });
});
