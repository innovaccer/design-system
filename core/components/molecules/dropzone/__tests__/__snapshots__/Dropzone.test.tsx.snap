// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropzone component snapshot 
  formatLabel: "Accepted formats: PDF, JPG", sizeLabel: "Maximum size: 30 MB", disabled: false, type: "compressed"
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--compressed Dropzone-animation Dropzone-animation--default"
    data-test="DesignSystem-Dropzone"
  >
    <svg
      class="Dropzone-icon--compressed"
      fill="none"
      viewBox="0 0 64 64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Dropzone-icon--default"
        d="M14.6667 53.3334C13.6 53.3334 12.6667 52.9334 11.8667 52.1334C11.0667 51.3334 10.6667 50.4 10.6667 49.3334V39.8H14.6667V49.3334H49.3333V39.8H53.3333V49.3334C53.3333 50.4 52.9333 51.3334 52.1333 52.1334C51.3333 52.9334 50.4 53.3334 49.3333 53.3334H14.6667ZM32 43.1334L19.1333 30.2667L22 27.4L30 35.4V10.6667H34V35.4L42 27.4L44.8667 30.2667L32 43.1334Z"
      />
    </svg>
    <div
      class="DropzoneWrapper DropzoneWrapper--compressed"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--default Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--link Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="0"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
      <span
        class="Text Text--subtle Text--regular"
        data-test="DesignSystem-Text"
      >
        Accepted formats: PDF, JPG
      </span>
      <span
        class="Text Text--subtle Text--regular"
        data-test="DesignSystem-Text"
      >
        Maximum size: 30 MB
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropzone component snapshot 
  formatLabel: "Accepted formats: PDF, JPG", sizeLabel: "Maximum size: 30 MB", disabled: false, type: "standard"
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--standard Dropzone-animation Dropzone-standard--default"
    data-test="DesignSystem-Dropzone"
  >
    <svg
      class="Dropzone-icon--standard"
      fill="none"
      viewBox="0 0 64 64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Dropzone-icon--default"
        d="M14.6667 53.3334C13.6 53.3334 12.6667 52.9334 11.8667 52.1334C11.0667 51.3334 10.6667 50.4 10.6667 49.3334V39.8H14.6667V49.3334H49.3333V39.8H53.3333V49.3334C53.3333 50.4 52.9333 51.3334 52.1333 52.1334C51.3333 52.9334 50.4 53.3334 49.3333 53.3334H14.6667ZM32 43.1334L19.1333 30.2667L22 27.4L30 35.4V10.6667H34V35.4L42 27.4L44.8667 30.2667L32 43.1334Z"
      />
    </svg>
    <div
      class="DropzoneWrapper DropzoneWrapper--standard"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--default Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--link Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="0"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
      <span
        class="Text Text--subtle Text--regular"
        data-test="DesignSystem-Text"
      >
        Accepted formats: PDF, JPG
      </span>
      <span
        class="Text Text--subtle Text--regular"
        data-test="DesignSystem-Text"
      >
        Maximum size: 30 MB
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropzone component snapshot 
  formatLabel: "Accepted formats: PDF, JPG", sizeLabel: "Maximum size: 30 MB", disabled: false, type: "tight"
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--tight Dropzone-animation Dropzone-animation--default"
    data-test="DesignSystem-Dropzone"
  >
    <div
      class="DropzoneWrapper DropzoneWrapper--tight"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--default Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--link Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="0"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
      <span
        class="Text Text--subtle Text--regular"
        data-test="DesignSystem-Text"
      >
        Accepted formats: PDF, JPG
      </span>
      <span
        class="Text Text--subtle Text--regular"
        data-test="DesignSystem-Text"
      >
        Maximum size: 30 MB
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropzone component snapshot 
  formatLabel: "Accepted formats: PDF, JPG", sizeLabel: "Maximum size: 30 MB", disabled: true, type: "compressed"
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--compressed Dropzone--disabled Dropzone-animation Dropzone-animation--default"
    data-test="DesignSystem-Dropzone"
  >
    <svg
      class="Dropzone-icon--compressed"
      fill="none"
      viewBox="0 0 64 64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Dropzone-icon--disabled"
        d="M14.6667 53.3334C13.6 53.3334 12.6667 52.9334 11.8667 52.1334C11.0667 51.3334 10.6667 50.4 10.6667 49.3334V39.8H14.6667V49.3334H49.3333V39.8H53.3333V49.3334C53.3333 50.4 52.9333 51.3334 52.1333 52.1334C51.3333 52.9334 50.4 53.3334 49.3333 53.3334H14.6667ZM32 43.1334L19.1333 30.2667L22 27.4L30 35.4V10.6667H34V35.4L42 27.4L44.8667 30.2667L32 43.1334Z"
      />
    </svg>
    <div
      class="DropzoneWrapper DropzoneWrapper--compressed"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--disabled Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--disabled Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="-1"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
      <span
        class="Text Text--disabled Text--regular"
        data-test="DesignSystem-Text"
      >
        Accepted formats: PDF, JPG
      </span>
      <span
        class="Text Text--disabled Text--regular"
        data-test="DesignSystem-Text"
      >
        Maximum size: 30 MB
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropzone component snapshot 
  formatLabel: "Accepted formats: PDF, JPG", sizeLabel: "Maximum size: 30 MB", disabled: true, type: "standard"
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--standard Dropzone--disabled Dropzone-animation Dropzone-standard--default"
    data-test="DesignSystem-Dropzone"
  >
    <svg
      class="Dropzone-icon--standard"
      fill="none"
      viewBox="0 0 64 64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Dropzone-icon--disabled"
        d="M14.6667 53.3334C13.6 53.3334 12.6667 52.9334 11.8667 52.1334C11.0667 51.3334 10.6667 50.4 10.6667 49.3334V39.8H14.6667V49.3334H49.3333V39.8H53.3333V49.3334C53.3333 50.4 52.9333 51.3334 52.1333 52.1334C51.3333 52.9334 50.4 53.3334 49.3333 53.3334H14.6667ZM32 43.1334L19.1333 30.2667L22 27.4L30 35.4V10.6667H34V35.4L42 27.4L44.8667 30.2667L32 43.1334Z"
      />
    </svg>
    <div
      class="DropzoneWrapper DropzoneWrapper--standard"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--disabled Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--disabled Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="-1"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
      <span
        class="Text Text--disabled Text--regular"
        data-test="DesignSystem-Text"
      >
        Accepted formats: PDF, JPG
      </span>
      <span
        class="Text Text--disabled Text--regular"
        data-test="DesignSystem-Text"
      >
        Maximum size: 30 MB
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropzone component snapshot 
  formatLabel: "Accepted formats: PDF, JPG", sizeLabel: "Maximum size: 30 MB", disabled: true, type: "tight"
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--tight Dropzone--disabled Dropzone-animation Dropzone-animation--default"
    data-test="DesignSystem-Dropzone"
  >
    <div
      class="DropzoneWrapper DropzoneWrapper--tight"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--disabled Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--disabled Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="-1"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
      <span
        class="Text Text--disabled Text--regular"
        data-test="DesignSystem-Text"
      >
        Accepted formats: PDF, JPG
      </span>
      <span
        class="Text Text--disabled Text--regular"
        data-test="DesignSystem-Text"
      >
        Maximum size: 30 MB
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropzone component snapshot 
  maxSize: 300, minSize: 50, preventDropOnDocument: false, multiple: false
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--standard Dropzone-animation Dropzone-standard--default"
    data-test="DesignSystem-Dropzone"
  >
    <svg
      class="Dropzone-icon--standard"
      fill="none"
      viewBox="0 0 64 64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Dropzone-icon--default"
        d="M14.6667 53.3334C13.6 53.3334 12.6667 52.9334 11.8667 52.1334C11.0667 51.3334 10.6667 50.4 10.6667 49.3334V39.8H14.6667V49.3334H49.3333V39.8H53.3333V49.3334C53.3333 50.4 52.9333 51.3334 52.1333 52.1334C51.3333 52.9334 50.4 53.3334 49.3333 53.3334H14.6667ZM32 43.1334L19.1333 30.2667L22 27.4L30 35.4V10.6667H34V35.4L42 27.4L44.8667 30.2667L32 43.1334Z"
      />
    </svg>
    <div
      class="DropzoneWrapper DropzoneWrapper--standard"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--default Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--link Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="0"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropzone component snapshot 
  maxSize: 300, minSize: 50, preventDropOnDocument: false, multiple: true
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--standard Dropzone-animation Dropzone-standard--default"
    data-test="DesignSystem-Dropzone"
  >
    <svg
      class="Dropzone-icon--standard"
      fill="none"
      viewBox="0 0 64 64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Dropzone-icon--default"
        d="M14.6667 53.3334C13.6 53.3334 12.6667 52.9334 11.8667 52.1334C11.0667 51.3334 10.6667 50.4 10.6667 49.3334V39.8H14.6667V49.3334H49.3333V39.8H53.3333V49.3334C53.3333 50.4 52.9333 51.3334 52.1333 52.1334C51.3333 52.9334 50.4 53.3334 49.3333 53.3334H14.6667ZM32 43.1334L19.1333 30.2667L22 27.4L30 35.4V10.6667H34V35.4L42 27.4L44.8667 30.2667L32 43.1334Z"
      />
    </svg>
    <div
      class="DropzoneWrapper DropzoneWrapper--standard"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--default Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--link Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="0"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropzone component snapshot 
  maxSize: 300, minSize: 50, preventDropOnDocument: true, multiple: false
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--standard Dropzone-animation Dropzone-standard--default"
    data-test="DesignSystem-Dropzone"
  >
    <svg
      class="Dropzone-icon--standard"
      fill="none"
      viewBox="0 0 64 64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Dropzone-icon--default"
        d="M14.6667 53.3334C13.6 53.3334 12.6667 52.9334 11.8667 52.1334C11.0667 51.3334 10.6667 50.4 10.6667 49.3334V39.8H14.6667V49.3334H49.3333V39.8H53.3333V49.3334C53.3333 50.4 52.9333 51.3334 52.1333 52.1334C51.3333 52.9334 50.4 53.3334 49.3333 53.3334H14.6667ZM32 43.1334L19.1333 30.2667L22 27.4L30 35.4V10.6667H34V35.4L42 27.4L44.8667 30.2667L32 43.1334Z"
      />
    </svg>
    <div
      class="DropzoneWrapper DropzoneWrapper--standard"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--default Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--link Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="0"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropzone component snapshot 
  maxSize: 300, minSize: 50, preventDropOnDocument: true, multiple: true
 1`] = `
<DocumentFragment>
  <div
    class="Dropzone Dropzone--standard Dropzone-animation Dropzone-standard--default"
    data-test="DesignSystem-Dropzone"
  >
    <svg
      class="Dropzone-icon--standard"
      fill="none"
      viewBox="0 0 64 64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="Dropzone-icon--default"
        d="M14.6667 53.3334C13.6 53.3334 12.6667 52.9334 11.8667 52.1334C11.0667 51.3334 10.6667 50.4 10.6667 49.3334V39.8H14.6667V49.3334H49.3333V39.8H53.3333V49.3334C53.3333 50.4 52.9333 51.3334 52.1333 52.1334C51.3333 52.9334 50.4 53.3334 49.3333 53.3334H14.6667ZM32 43.1334L19.1333 30.2667L22 27.4L30 35.4V10.6667H34V35.4L42 27.4L44.8667 30.2667L32 43.1334Z"
      />
    </svg>
    <div
      class="DropzoneWrapper DropzoneWrapper--standard"
      data-test="DesignSystem-Dropzone-Wrapper"
    >
      <span>
        <span
          class="Text Text--default Text--strong Text--large mr-2"
          data-test="DesignSystem-Text"
        >
          Drag your files here or
        </span>
        <span
          aria-label="Drag your files here or click to select files"
          class="Text Text--link Text--strong Text--large ml-2 cursor-pointer"
          data-test="DesignSystem-Text"
          role="button"
          tabindex="0"
        >
          browse files
        </span>
        <input
          autocomplete="off"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
